lib/astutils.cpp

```diff
@@ -2347,6 +2347,10 @@ bool isVariableChangedByFunctionCall(const Token *tok, int indirect, const Setti
     if (addressOf)
         indirect++;
 
+    const bool deref = tok->astParent() && tok->astParent()->isUnaryOp("*");
+    if (deref)
```

Comment:
would it make sense to add a `indirect > 0` check to avoid that it will be negative?

Reply:
Check added.